CMakeLists.txt

```diff
@@ -196,6 +196,17 @@ add_library(
     ${libslate_src}
 )
 
+#--------------------
+# matgen
+add_library(
+    slate_matgen
+    matgen/generate_matrix_ge.cc
+    matgen/generate_matrix_he_and_tz.cc
+    matgen/generate_matrix_utils.cc
+    src/set_lambdas.cc
```

Comment:
We don't need to compile `src/set_lambda.cc` as part of the matgen library.  [Everything in `src` gets compiled as part of libslate.](https://github.com/icl-utk-edu/slate/blob/master/CMakeLists.txt#L184)

Reply:
Originally I was getting an error when building when I did not place `src/set_lambdas.cc` within the `CMakeLists.txt` file. However, I believe putting the forward declaration within the `include/slate/slate.hh` fixes this, because I just tested it and it now works after making those two changes. 

Why would we need to add `generate_matrix.hh` to `CMakeLists.txt`?  Would we not want to add that within `include/slate/slate.hh` in the form of `#include "generate_matrix.hh"`? This would be similar to how the simplified c++ api header file is included/loaded. 

I believe doing that would remove needing the following include in `test.cc`:
```
#include "../matgen/generate_matrix.hh"
```

However, I do not believe this is best practice.

Reply:
CMake installs everything that is in the include/ directory:
```
# Install header files
install(
    # / copies contents, not directory itself
    DIRECTORY "${PROJECT_SOURCE_DIR}/include/"
    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
    FILES_MATCHING REGEX "\\.(h|hh)$"
)
```
so as long as `generate_matrix.hh` is in `include/slate/`, that should work.

Right, we should avoid use `../` in include paths. (There are some cases for doing this, but it's generally discouraged.) So whatever struct gets passed in to the generator should be defined in `generate_matrix.hh`.

Reply:
With the last push that I did, I have moved the `generate_matrix.hh` file into `include/slate/` and updated the rest of the files appropriately to have the right `#include`.